sphinx/writers/latex.py

```diff
@@ -424,21 +424,32 @@ def escape_abbr(text):
     return re.sub(r'\.(?=\s|$)', r'.\@', text)
 
 
-def rstdim_to_latexdim(width_str):
-    # type: (unicode) -> unicode
+def rstdim_to_latexdim(width_str, scale = 100):
+    # type: (unicode, int) -> unicode
     """Convert `width_str` with rst length to LaTeX length."""
     match = re.match(r'^(\d*\.?\d*)\s*(\S*)$', width_str)
     if not match:
         raise ValueError
     res = width_str
     amount, unit = match.groups()[:2]
-    float(amount)  # validate amount is float
-    if unit in ('', "px"):
-        res = "%s\\sphinxpxdimen" % amount
-    elif unit == 'pt':
-        res = '%sbp' % amount  # convert to 'bp'
-    elif unit == "%":
-        res = "%.3f\\linewidth" % (float(amount) / 100.0)
+    if scale == 100:
+        float(amount)  # validate amount is float
+        if unit in ('', "px"):
+            res = "%s\\sphinxpxdimen" % amount
+        elif unit == 'pt':
+            res = '%sbp' % amount  # convert to 'bp'
+        elif unit == "%":
+            res = "%.3f\\linewidth" % (float(amount) / 100.0)
+    else:
+        amount_float = float(amount) * scale / 100.0
```

Comment:
mypy complained when I used amount = float(amount) etc..., and then in all next, because it is not able to recognize the new type assigned to variable; in my mind this is deficiency of mypy, although I am aware using same identifier with two distinct types might not be best taste.